docs/rfc/rfc-005-authentication.md

```diff
@@ -0,0 +1,132 @@
+}
+message MsgCreateAuthenticatedAccountResponse {
+  accounts.MsgDeployResponse deploy_response = 1;
+}
+message MsgUpdateCredentials {
+  string sender = 1;
+  string kind = 2;
+  google.Protobuf.Any new_credential = 3;
+}
+message MsgUpdateCredentialsResponse {}
+message MsgDeleteCredentials {
+  string sender = 1;
+}
+message MsgDeleteCredentialsResponse {}
+```
+
+### MsgCreateAuthenticatedAccount
+
+This message contains an opaque credential defined as `google.Protobuf.Any`, alongside an `accounts.MsgDeploy` request.
+This creates a new account and couples it with a credential.
+
+### MsgUpdateCredentials & MsgDeleteCredentials
+
+The former allows for credentials of an account to be updated the latter destroys credentials for an account, making it
+effectively impossible for the account to send state transitions from a TX forever, unless the account has logic to again
+update its credentials.
+
+### The credential interface
+
+The credential interface, represented in our gRPC `tx` interface as a `google.Protobuf.Any`, is implemented by any type
+which satisfies the following interface:
+
+```go
+package authn
+type Credential[T any, PT interface{ *T; proto.Message }] interface {
+	*T
+	VerifySignedBytes(msgBytes []byte, signature []byte) bool
+}
+```
+#### VerifySignedBytes
+
+The credential is fetched from state based on the entity trying to authenticate the `state transition`, we know this by
+pulling the signer of the message from the message itself (currently). Then the credential just applies the verification
+logic.
+
+### Further discussion
+
+#### Credentials abstraction
+
+Credentials are currently abstracted over entities that can verify arbitrary signed bytes, which covers the use-case of crypto
+curves.
+
+The idea is that in the future we can further abstract the authentication mechanism (not only the curve), this means that a chain
+can be able to define its own authentication mechanisms, which are not tied to the `SignMode` provided by the sdk.
+Trying to fit this change right now would have yielded into a much broader work, that would have most likely lead to impactful
+breaking changes.
+
+#### Update x/auth, instead of creating an x/authn module
+
+Considering the limitations over the credentials abstraction, the changeset for `x/auth` is still limited, so we could
+just update `x/auth` to support `x/accounts` deployment and key rotation messages.
+
+## Migration
+
+### Phase1
+
+- Implement `x/accounts` and `x/authn`
+- Accounts can be migrated explicitly either by sending `x/auth` a `MsgMigrateToAuthn`, or implicitly in the `AnteHandler`.
+- Move `auth` permissions  from `x/auth` to `x/bank`, as bank is the only consumer of this information.
+
+## Major implications
+
+- Account unique identifier (represented as AccAddress) is decoupled from the authentication mechanism.
+- Accounts need to always be explicitly created. Bank would not create an account in case it doesn't exist during a `MsgSend` execution.
```

Comment:
This has negative implications for IBC where you would have to create your account on a chain before being able to IBC to it, breaking atomicity. 

Some account types may require an explicit deploy message, but I think for most (the ones that support an empty deploy msg) we can just ensure they are always created when on first interaction

Reply:
for ibc they would use the normal account system that exists today,